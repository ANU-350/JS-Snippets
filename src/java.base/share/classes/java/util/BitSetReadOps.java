/*
 * Copyright (c) 1995, 2023, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Oracle designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Oracle in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

package java.util;

import java.util.stream.IntStream;

/**
 * This interface models the read operations of a {@linkplain BitSet}.
 *
 * @since 22
 */
// Todo: Come up with a better name.
public sealed interface BitSetReadOps
        extends Cloneable
        permits BitSet, BitSet.OfImmutable {

    /**
     * Returns a new byte array containing all the bits in this bit set.
     *
     * <p>More precisely, if
     * <br>{@code byte[] bytes = s.toByteArray();}
     * <br>then {@code bytes.length == (s.length()+7)/8} and
     * <br>{@code s.get(n) == ((bytes[n/8] & (1<<(n%8))) != 0)}
     * <br>for all {@code n < 8 * bytes.length}.
     *
     * @return a byte array containing a little-endian representation
     *         of all the bits in this bit set
     * @since 1.7
     */
    byte[] toByteArray();

    /**
     * Returns a new long array containing all the bits in this bit set.
     *
     * <p>More precisely, if
     * <br>{@code long[] longs = s.toLongArray();}
     * <br>then {@code longs.length == (s.length()+63)/64} and
     * <br>{@code s.get(n) == ((longs[n/64] & (1L<<(n%64))) != 0)}
     * <br>for all {@code n < 64 * longs.length}.
     *
     * @return a long array containing a little-endian representation
     *         of all the bits in this bit set
     * @since 1.7
     */
    long[] toLongArray();

    /**
     * Returns the value of the bit with the specified index. The value
     * is {@code true} if the bit with the index {@code bitIndex}
     * is currently set in this {@code BitSet}; otherwise, the result
     * is {@code false}.
     *
     * @param  bitIndex   the bit index
     * @return the value of the bit with the specified index
     * @throws IndexOutOfBoundsException if the specified index is negative
     */
    boolean get(int bitIndex);

    /**
     * Returns a new {@code BitSet} composed of bits from this {@code BitSet}
     * from {@code fromIndex} (inclusive) to {@code toIndex} (exclusive).
     *
     * @param  fromIndex index of the first bit to include
     * @param  toIndex index after the last bit to include
     * @return a new {@code BitSet} from a range of this {@code BitSet}
     * @throws IndexOutOfBoundsException if {@code fromIndex} is negative,
     *         or {@code toIndex} is negative, or {@code fromIndex} is
     *         larger than {@code toIndex}
     * @since  1.4
     */
    BitSetReadOps get(int fromIndex, int toIndex);

    /**
     * Returns the index of the first bit that is set to {@code true}
     * that occurs on or after the specified starting index. If no such
     * bit exists then {@code -1} is returned.
     *
     * <p>To iterate over the {@code true} bits in a {@code BitSet},
     * use the following loop:
     *
     *  <pre> {@code
     * for (int i = bs.nextSetBit(0); i >= 0; i = bs.nextSetBit(i+1)) {
     *     // operate on index i here
     *     if (i == Integer.MAX_VALUE) {
     *         break; // or (i+1) would overflow
     *     }
     * }}</pre>
     *
     * @param  fromIndex the index to start checking from (inclusive)
     * @return the index of the next set bit, or {@code -1} if there
     *         is no such bit
     * @throws IndexOutOfBoundsException if the specified index is negative
     * @since  1.4
     */
    int nextSetBit(int fromIndex);

    /**
     * Returns the index of the first bit that is set to {@code false}
     * that occurs on or after the specified starting index.
     *
     * @param  fromIndex the index to start checking from (inclusive)
     * @return the index of the next clear bit
     * @throws IndexOutOfBoundsException if the specified index is negative
     * @since  1.4
     */
    int nextClearBit(int fromIndex);

    /**
     * Returns the index of the nearest bit that is set to {@code true}
     * that occurs on or before the specified starting index.
     * If no such bit exists, or if {@code -1} is given as the
     * starting index, then {@code -1} is returned.
     *
     * <p>To iterate over the {@code true} bits in a {@code BitSet},
     * use the following loop:
     *
     *  <pre> {@code
     * for (int i = bs.length(); (i = bs.previousSetBit(i-1)) >= 0; ) {
     *     // operate on index i here
     * }}</pre>
     *
     * @param  fromIndex the index to start checking from (inclusive)
     * @return the index of the previous set bit, or {@code -1} if there
     *         is no such bit
     * @throws IndexOutOfBoundsException if the specified index is less
     *         than {@code -1}
     * @since  1.7
     */
    int previousSetBit(int fromIndex);

    /**
     * Returns the index of the nearest bit that is set to {@code false}
     * that occurs on or before the specified starting index.
     * If no such bit exists, or if {@code -1} is given as the
     * starting index, then {@code -1} is returned.
     *
     * @param  fromIndex the index to start checking from (inclusive)
     * @return the index of the previous clear bit, or {@code -1} if there
     *         is no such bit
     * @throws IndexOutOfBoundsException if the specified index is less
     *         than {@code -1}
     * @since  1.7
     */
    int previousClearBit(int fromIndex);

    /**
     * Returns the "logical size" of this {@code BitSet}: the index of
     * the highest set bit in the {@code BitSet} plus one. Returns zero
     * if the {@code BitSet} contains no set bits.
     *
     * @return the logical size of this {@code BitSet}
     * @since  1.2
     */
    int length();

    /**
     * Returns true if this {@code BitSet} contains no bits that are set
     * to {@code true}.
     *
     * @return boolean indicating whether this {@code BitSet} is empty
     * @since  1.4
     */
    boolean isEmpty();

    /**
     * Returns true if the specified {@code BitSet} has any bits set to
     * {@code true} that are also set to {@code true} in this {@code BitSet}.
     *
     * @param  set {@code BitSet} to intersect with
     * @return boolean indicating whether this {@code BitSet} intersects
     *         the specified {@code BitSet}
     * @since  1.4
     */
    boolean intersects(BitSet set);

    /**
     * Returns true if the specified {@code BitSet} has any bits set to
     * {@code true} that are also set to {@code true} in this {@code BitSet}.
     *
     * @param  set {@code BitSet} to intersect with
     * @return boolean indicating whether this {@code BitSet} intersects
     *         the specified {@code BitSet}
     * @since  22
     */
    boolean intersects(BitSetReadOps set);

    /**
     * Returns the number of bits set to {@code true} in this {@code BitSet}.
     *
     * @return the number of bits set to {@code true} in this {@code BitSet}
     * @since  1.4
     */
    int cardinality();

    /**
     * {@return the hash code value for this bit set}
     *
     * The hash code depends only on which bits are set within this
     * {@code BitSet}.
     *
     * <p>The hash code is defined to be the result of the following
     * calculation:
     *  <pre> {@code
     * public int hashCode() {
     *     long h = 1234;
     *     long[] words = toLongArray();
     *     for (int i = words.length; --i >= 0; )
     *         h ^= words[i] * (i + 1);
     *     return (int)((h >> 32) ^ h);
     * }}</pre>
     * Note that the hash code changes if the set of bits is altered.
     */
    @Override
    int hashCode();

    /**
     * Returns the number of bits of space actually in use by this
     * {@code BitSet} to represent bit values.
     * The maximum element in the set is the size - 1st element.
     *
     * @return the number of bits currently in this bit set
     */
    int size();

    /**
     * Compares this bit set against the specified object.
     * The result is {@code true} if and only if the argument is
     * not {@code null} and is a {@code BitSet} object that has
     * exactly the same set of bits set to {@code true} as this bit
     * set. That is, for every nonnegative {@code int} index {@code k},
     * <pre>((BitSet)obj).get(k) == this.get(k)</pre>
     * must be true. The current sizes of the two bit sets are not compared.
     *
     * @param  obj the object to compare with
     * @return {@code true} if the objects are the same;
     *         {@code false} otherwise
     * @see    #size()
     */
    @Override
    boolean equals(Object obj);

    /**
     * Cloning this {@code BitSet} produces a new {@code BitSet}
     * that is equal to it.
     * The clone of the bit set is another bit set that has exactly the
     * same bits set to {@code true} as this bit set.
     *
     * @return a clone of this bit set
     * @see    #size()
     */
    Object clone();

    /**
     * Returns a string representation of this bit set. For every index
     * for which this {@code BitSet} contains a bit in the set
     * state, the decimal representation of that index is included in
     * the result. Such indices are listed in order from lowest to
     * highest, separated by ",&nbsp;" (a comma and a space) and
     * surrounded by braces, resulting in the usual mathematical
     * notation for a set of integers.
     *
     * <p>Example:
     * <pre>
     * BitSet drPepper = new BitSet();</pre>
     * Now {@code drPepper.toString()} returns "{@code {}}".
     * <pre>
     * drPepper.set(2);</pre>
     * Now {@code drPepper.toString()} returns "{@code {2}}".
     * <pre>
     * drPepper.set(4);
     * drPepper.set(10);</pre>
     * Now {@code drPepper.toString()} returns "{@code {2, 4, 10}}".
     *
     * @return a string representation of this bit set
     */
    @Override
    String toString();

    /**
     * Returns a stream of indices for which this {@code BitSet}
     * contains a bit in the set state. The indices are returned
     * in order, from lowest to highest. The size of the stream
     * is the number of bits in the set state, equal to the value
     * returned by the {@link #cardinality()} method.
     *
     * <p>The stream binds to this bit set when the terminal stream operation
     * commences (specifically, the spliterator for the stream is
     * <a href="Spliterator.html#binding"><em>late-binding</em></a>).  If the
     * bit set is modified during that operation then the result is undefined.
     *
     * @return a stream of integers representing set indices
     * @since 1.8
     */
    IntStream stream();

}
