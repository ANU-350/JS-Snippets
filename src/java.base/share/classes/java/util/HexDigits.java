/*
 * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Oracle designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Oracle in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

package java.util;

import java.lang.invoke.MethodHandle;

import jdk.internal.vm.annotation.Stable;

/**
 * Digits class for hexadecimal digits.
 *
 * @since 21
 */
final class HexDigits implements Digits {
    @Stable
    private static final short[] DIGITS;

    /**
     * Singleton instance of HexDigits.
     */
    static final Digits INSTANCE = new HexDigits();

    static {
        short[] digits = new short[16 * 16];

        for (int i = 0; i < 16; i++) {
            short hi = (short) (i < 10 ? i + '0' : i - 10 + 'a');

            for (int j = 0; j < 16; j++) {
                short lo = (short) ((j < 10 ? j + '0' : j - 10 + 'a') << 8);
                digits[(i << 4) + j] = (short) (hi | lo);
            }
        }

        DIGITS = digits;
    }

    /**
     * Constructor.
     */
    private HexDigits() {
    }

    /**
     * Return a big-endian packed integer for the 4 ASCII bytes for an input unsigned 2-byte integer.
     * {@code b0} is the most significant byte and {@code b1} is the least significant byte.
     * The integer is passed byte-wise to allow reordering of execution.
     */
    static int packDigits(int b0, int b1) {
        return DIGITS[b0 & 0xff] | (DIGITS[b1 & 0xff] << 16);
    }

    /**
     * Return a big-endian packed long for the 8 ASCII bytes for an input unsigned 4-byte integer.
     * {@code b0} is the most significant byte and {@code b3} is the least significant byte.
     * The integer is passed byte-wise to allow reordering of execution.
     */
    static long packDigits(int b0, int b1, int b2, int b3) {
        return DIGITS[b0 & 0xff]
                | (DIGITS[b1 & 0xff] << 16)
                | (((long) DIGITS[b2 & 0xff]) << 32)
                | (((long) DIGITS[b3 & 0xff]) << 48);
    }

    @Override
    public int digits(long value, byte[] buffer, int index,
                      MethodHandle putCharMH) throws Throwable {
        while ((value & ~0xFF) != 0) {
            int digits = DIGITS[(int) (value & 0xFF)];
            value >>>= 8;
            putCharMH.invokeExact(buffer, --index, digits >> 8);
            putCharMH.invokeExact(buffer, --index, digits & 0xFF);
        }

        int digits = DIGITS[(int) (value & 0xFF)];
        putCharMH.invokeExact(buffer, --index, digits >> 8);

        if (0xF < value) {
            putCharMH.invokeExact(buffer, --index, digits & 0xFF);
        }

        return index;
    }

    @Override
    public int size(long value) {
        return value == 0 ? 1 :
                67 - Long.numberOfLeadingZeros(value) >> 2;
    }
}
