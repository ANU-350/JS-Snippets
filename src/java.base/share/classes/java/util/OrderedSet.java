/*
 * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Oracle designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Oracle in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

package java.util;

/**
 * A set that has a well-defined order. Provides methods for
 * convenient access to elements at both ends, as well as a reversed-order view.
 *
 * @param <E> the type of elements in this collection
 * @since XXX
 */
public interface OrderedSet<E> extends Set<E>, OrderedCollection<E> {
    /**
     * Throws UnsupportedOperationException.
     * @param e the element to be added
     * @throws UnsupportedOperationException always
     */
    default void addFirst(E e) { throw new UnsupportedOperationException(); }

    /**
     * Throws UnsupportedOperationException.
     * @param e the element to be added
     * @throws UnsupportedOperationException always
     */
    default void addLast(E e) { throw new UnsupportedOperationException(); }

    /**
     * Returns a reversed-order view of this set. If the implementation
     * permits modifications to this view, the modifications "write through"
     * to the underlying collection. Depending upon the implementation's
     * concurrent modification policy, changes to the underlying collection
     * may be visible in this reversed view.
     * @return a reversed-order view
     */
    OrderedSet<E> reversed();
}
