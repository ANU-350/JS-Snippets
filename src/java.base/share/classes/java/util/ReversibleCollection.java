/*
 * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Oracle designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Oracle in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

package java.util;

/**
 * A linear collection that has a well-defined order and that is reversible.
 * Provides methods for a reverse-ordered view as well as convenient access
 * to elements at both ends.
 *
 * @param <E> the type of elements in this collection
 * @since XXX
 */
public interface ReversibleCollection<E> extends Collection<E> {
    /**
     * Returns a reversed-order view of this collection. If the implementation
     * permits modifications to this view, the modifications "write through"
     * to the underlying collection. Depending upon the implementation's
     * concurrent modification policy, changes to the underlying collection
     * may be visible in this reversed view.
     * @return a reversed-order view of this collection
     */
    ReversibleCollection<E> reversedCollection();

    /**
     * Adds an element at the front of this collection (optional operation).
     * @param e the element to be added
     * @throws NullPointerException if the specified element is null and this
     *         collection does not permit null elements
     */
    void addFirst(E e);

    /**
     * Adds an element at the end of this collection (optional operation).
     * Synonymous with add(E).
     * @param e the element to be added.
     * @throws NullPointerException if the specified element is null and this
     *         collection does not permit null elements
     */
    void addLast(E e);

    /**
     * Gets the element at the front of this collection.
     * @return the retrieved element
     * @throws NoSuchElementException if this collection is empty
     */
    E getFirst();

    /**
     * Gets the element at the end of this collection.
     * @return the retrieved element
     * @throws NoSuchElementException if this collection is empty
     */
    E getLast();

    /**
     * Removes and returns the first element of this collection (optional operation).
     * @return the removed element
     * @throws NoSuchElementException if this collection is empty
     */
    E removeFirst();

    /**
     * Removes and returns the last element of this collection (optional operation).
     * @return the removed element
     * @throws NoSuchElementException if this collection is empty
     */
    E removeLast();
}
