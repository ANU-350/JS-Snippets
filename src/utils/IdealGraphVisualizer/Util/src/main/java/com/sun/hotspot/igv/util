../../../../../../com/sun/hotspot/igv/util