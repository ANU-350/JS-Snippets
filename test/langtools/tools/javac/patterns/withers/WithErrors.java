/*
 * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

/**
 * @test
 * @bug 8324651
 * @summary Support for derived record creation expression
 * @enablePreview
 * @compile/fail/ref=WithErrors.out -XDrawDiagnostics -XDshould-stop.at=FLOW WithErrors.java
 */

import java.util.List;

public class WithErrors {

    private int field;

    private void assignments(R input) {
        int i = 0;

        input = input with {
            int l;

            value = "nue"; //OK - assignment to the component
            i = ++i + i++ + --i + i-- + -i + +i; //error - assignment to outter variable
            i += 1; //error - (compound) assignment to outter variable
            field = 0; //error - unqualified assignment to field
            this.field = 0; // OK - qualified assignment
            l = 0; //OK - assignment to a variable local to the block
        };

        input = input with {
            int l1;
            Runnable _ = () -> {
                int l2;

                value = "nue"; //error - cannot assign inside lambda
                i = 0; //error - assignment to outter variable, and inside lambda
                i += 1; //error - (compound) assignment to outter variable
                field = 0; //error - unqualified assignment to field
                this.field = 0; // OK - qualified assignment
                l1 = 0; //error - cannot assign inside lambda
                l2 = 0; //OK - assignment to a variable local to the block
            };
        };

        input = input with {
            Runnable _ = () -> {
                String _ = value; //error - "value" is not effectivelly final
            };
            value = "nue";
        };

        input = input with {
            Runnable _ = () -> {
                String _ = value; //OK - "value" is effectivelly final
            };
        };

        input = input with {
            int l1;
            Runnable _ = new Runnable() {
                public void run () {
                    int l2;

                    value = "nue"; //error - cannot assign inside the anonymous class
                    i = 0; //error - cannot assign inside the anonymous class
                    field = 0; //OK - assignment to outer field from inside the anonymous class
                    WithErrors.this.field = 0; // OK - qualified assignment
                    l1 = 0; //error - cannot assign inside lambda
                    l2 = 0; //OK - assignment to a variable local to this block/method
                }
            };
        };

        input = input with {
            Runnable _ = new Runnable() {
                public void run () {
                    String _ = value; //error - "value" is not effectivelly final
                }
            };
            value = "nue";
        };

        input = input with {
            Runnable _ = new Runnable() {
                public void run () {
                    String _ = value; //OK - "value" is effectivelly final
                }
            };
        };

        String _ = "" with {};
        int _ = 1 with {};
    }

    private void controlFlow(R input) {
        if (true) {
            input = input with {
                return ;
            };
        }

        for (;;) {
            input = input with {
                break;
            };
        }

        input = input with {
            for (;;) {
                break;
            }
        };

        for (String s : List.of("")) {
            input = input with {
                break;
            };
        }

        input = input with {
            for (String s : List.of("")) {
                break;
            }
        };

        while (true) {
            input = input with {
                break;
            };
        }

        input = input with {
            while (true) {
                break;
            }
        };

        do {
            input = input with {
                break;
            };
        } while (true);

        input = input with {
            do {
                break;
            } while (true);
        };

        switch (0) {
            default ->
                input = input with {
                    break;
                };
        }

        input = input with {
            switch (0) {
                default -> {break;}
            }
        };

        if (true) {
            for (;;) {
                input = input with {
                    continue;
                };
            }
        }

        if (true) {
            input = input with {
                for (;;) {
                    continue;
                }
            };
        }

        if (true) {
            for (String s : List.of("")) {
                input = input with {
                    continue;
                };
            }
        }

        if (true) {
            input = input with {
                for (String s : List.of("")) {
                    continue;
                }
            };
        }

        if (true) {
            while (true) {
                input = input with {
                    continue;
                };
            }
        }

        if (true) {
            input = input with {
                while (true) {
                    continue;
                }
            };
        }

        if (true) {
            do {
                input = input with {
                    continue;
                };
            } while (true);
        }

        if (true) {
            input = input with {
                do {
                    continue;
                } while (true);
            };
        }

        int _ = switch (0) {
            default -> {
                input = input with {
                    yield 0;
                };
            }
        };

        input = input with {
            int _ = switch (0) {
                default -> { yield 0; }
            };
        };
    }

    record R(String value) {}

}
