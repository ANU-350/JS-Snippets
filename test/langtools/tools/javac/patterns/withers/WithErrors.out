WithErrors.java:42:13: compiler.err.with.assignment.not.allowed: i
WithErrors.java:43:13: compiler.err.with.assignment.not.allowed: field
WithErrors.java:54:17: compiler.err.with.assignment.not.allowed: i
WithErrors.java:55:17: compiler.err.with.assignment.not.allowed: field
WithErrors.java:108:23: compiler.err.derived.expression.no.record
WithErrors.java:109:19: compiler.err.derived.expression.no.record
WithErrors.java:115:17: compiler.err.with.return.not.allowed
WithErrors.java:121:17: compiler.err.with.break.not.allowed
WithErrors.java:133:17: compiler.err.with.break.not.allowed
WithErrors.java:145:17: compiler.err.with.break.not.allowed
WithErrors.java:157:17: compiler.err.with.break.not.allowed
WithErrors.java:170:21: compiler.err.with.break.not.allowed
WithErrors.java:183:21: compiler.err.with.continue.not.allowed
WithErrors.java:199:21: compiler.err.with.continue.not.allowed
WithErrors.java:215:21: compiler.err.with.continue.not.allowed
WithErrors.java:231:21: compiler.err.with.continue.not.allowed
WithErrors.java:247:21: compiler.err.with.yield.not.allowed
WithErrors.java:53:17: compiler.err.cant.ref.non.effectively.final.var: value, (compiler.misc.lambda)
WithErrors.java:57:17: compiler.err.cant.ref.non.effectively.final.var: l1, (compiler.misc.lambda)
WithErrors.java:64:28: compiler.err.cant.ref.non.effectively.final.var: value, (compiler.misc.lambda)
WithErrors.java:71:28: compiler.err.cant.ref.non.effectively.final.var: value, (compiler.misc.lambda)
WithErrors.java:81:21: compiler.err.cant.ref.non.effectively.final.var: value, (compiler.misc.inner.cls)
WithErrors.java:82:21: compiler.err.cant.ref.non.effectively.final.var: i, (compiler.misc.inner.cls)
WithErrors.java:85:21: compiler.err.cant.ref.non.effectively.final.var: l1, (compiler.misc.inner.cls)
WithErrors.java:94:32: compiler.err.cant.ref.non.effectively.final.var: value, (compiler.misc.inner.cls)
WithErrors.java:103:32: compiler.err.cant.ref.non.effectively.final.var: value, (compiler.misc.inner.cls)
26 errors
